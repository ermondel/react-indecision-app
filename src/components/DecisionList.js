import React from 'react';
import DecisionItem from './DecisionItem';

const DecisionList = ({ options, removeItem }) => {
  if (options.length <= 0) {
    return (
      <p className='decisions__list__message'>
        Please add an option to get started!
      </p>
    );
  }

  return options.map((option, index) => (
    <DecisionItem key={option} value={option} num={index + 1} remove={removeItem} />
  ));
};

export default DecisionList;
