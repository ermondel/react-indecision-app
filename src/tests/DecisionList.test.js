import React from 'react';
import renderer from 'react-test-renderer';
import DecisionList from '../components/DecisionList';
import decisions from './fixtures/decisions';

it('<DecisionList /> without data', () => {
  const tree = renderer.create(<DecisionList decisions={[]} removeDecision={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('<DecisionList /> with data', () => {
  const tree = renderer.create(<DecisionList decisions={decisions} removeDecision={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});
