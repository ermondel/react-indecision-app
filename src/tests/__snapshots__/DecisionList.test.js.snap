// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DecisionList /> with data 1`] = `
Array [
  <div
    className="decisions__item"
  >
    <p
      className="decisions__value"
    >
      1
      . 
      The pen is mightier than the sword.
    </p>
    <button
      className="decisions__remove-btn"
      onClick={[Function]}
    >
      remove
    </button>
  </div>,
  <div
    className="decisions__item"
  >
    <p
      className="decisions__value"
    >
      2
      . 
      When in Rome, do as the Romans.
    </p>
    <button
      className="decisions__remove-btn"
      onClick={[Function]}
    >
      remove
    </button>
  </div>,
  <div
    className="decisions__item"
  >
    <p
      className="decisions__value"
    >
      3
      . 
      The squeaky wheel gets the grease.
    </p>
    <button
      className="decisions__remove-btn"
      onClick={[Function]}
    >
      remove
    </button>
  </div>,
  <div
    className="decisions__item"
  >
    <p
      className="decisions__value"
    >
      4
      . 
      When the going gets tough, the tough get going.
    </p>
    <button
      className="decisions__remove-btn"
      onClick={[Function]}
    >
      remove
    </button>
  </div>,
  <div
    className="decisions__item"
  >
    <p
      className="decisions__value"
    >
      5
      . 
      Hope for the best, but prepare for the worst.
    </p>
    <button
      className="decisions__remove-btn"
      onClick={[Function]}
    >
      remove
    </button>
  </div>,
]
`;

exports[`<DecisionList /> without data 1`] = `
<p
  className="decisions__list__message"
>
  Please add an option to get started!
</p>
`;
